mars-sim-core/src/main/java/com/mars_sim/core/vehicle/Vehicle.java

```diff
@@ -1776,109 +1773,85 @@ public SalvageInfo getSalvageInfo() {
-			if (garage != null) {
-				centerLoc = garage.getPosition();
+			Building nearbyGarage = BuildingManager.getAGarage(getSettlement());
+			if (nearbyGarage != null) {
+				centerLoc = nearbyGarage.getPosition();
 			}
 		}
+		placeVehicleNearSettlementCenter(centerLoc);
+	}
 
-		// Place the vehicle starting from the settlement center (0,0).
-
+	private void placeVehicleNearSettlementCenter(LocalPosition centerLoc) {
 		int oX = 10;
-		int oY = 10;
+        int step = 2;
 		double newFacing = 0D;
 		LocalPosition newLoc = null;
-		int step = 2;
 		boolean foundGoodLocation = false;
+		boolean isSmallVehicle = VehicleType.isDrone(getVehicleType()) || getVehicleType() == VehicleType.LUV;
 
-		boolean isSmallVehicle = VehicleType.isDrone(getVehicleType())
-				|| getVehicleType() == VehicleType.LUV;
-
-		double d = VEHICLE_CLEARANCE_0;
-		if (isSmallVehicle)
-			d = VEHICLE_CLEARANCE_1;
-		
-		// Note: enlarge (times 1.25) the dimension of a vehicle to avoid getting 
-		// trapped within those enclosed space surrounded by buildings or hallways.
-		
-		double w = getWidth() * d * 1.25;
+		double d = isSmallVehicle ? VEHICLE_CLEARANCE_1 : VEHICLE_CLEARANCE_0;
+		double w = getWidth() * d * 1.25; // enlarge (times 1.25) to avoid  getting trapped
 		double l = getLength() * d * 1.25;
-		
-		// Note: May need a more permanent solution by figuring out how to detect those enclosed space
-		
+
+		int maxDistance = 500;
+		int maxAttemptsPerRange = 10;
 		int count = 0;
-		
-		// Try iteratively outward from 10m to 500m distance range.
-		for (int x = oX; (x < 500) && !foundGoodLocation; x+=step) {
+
+		for (int x = oX; (x < maxDistance) && !foundGoodLocation; x+=step) {
 			// Try random locations at each distance range.
-			for (int y = oY; (y < 500) && !foundGoodLocation; y++) {
-				double distance = Math.max(y, RandomUtil.getRandomDouble(-.5*x, .5*x) + .5*y);
+			for (int attempts = 0; attempts < maxAttemptsPerRange; attempts++) {
+				double distance = RandomUtil.getRandomDouble(x * 0.5, x);
 				double radianDirection = RandomUtil.getRandomDouble(Math.PI * 2D);
-				
+
 				newLoc = centerLoc.getPosition(distance, radianDirection);
 				newFacing = RandomUtil.getRandomDouble(360D);
 
-				// Check if new vehicle location collides with anything.
-				
-				// Note: excessive calling increase CPU Util
 				foundGoodLocation = LocalAreaUtil.isObjectCollisionFree(this, w, l,
-								newLoc.getX(), newLoc.getY(), 
-								newFacing, getCoordinates());
-				
+						newLoc.getX(), newLoc.getY(),
+						newFacing, getCoordinates());
+
 				count++;
 			}
 		}
 
 		if (foundGoodLocation) {
 			setParkedLocation(newLoc, newFacing);
-			logger.info(this, "Moved to new parking loc at (" 
-					+ Math.round(newLoc.getX() * 10.0)/10.0 
-					+ ", " + Math.round(newLoc.getY() * 10.0)/10.0 
+			logger.info(this, "Moved to new parking loc at ("
+					+ Math.round(newLoc.getX() * maxAttemptsPerRange)/maxAttemptsPerRange
```

Comment:
Math.round(abc * 10.0)/10.0 is a common way of deriving a double with just 1 decimal place.

Math.round(abc * 10)/10 is not the same as Math.round(abc * 10)/10. 

In this situation, the effect is not severe since it's only printing the result of a new x and y coordinate.

Just so you know.